packages/medusa/src/loaders/medusa-app.ts

```diff
@@ -94,6 +100,30 @@ export const loadMedusaApp = async (
     injectedDependencies,
   })
 
+  const requiredModuleKeys = [Modules.PRODUCT, Modules.PRICING]
```

Comment:
question: Are we sure we want to force that? Couldn't the flag be used for things unrelated to the modules and that are not required? I think that if the module is required we are checking that and warn the user in places it is required such as using the workflows in the end point?

Reply:
I found that to be a very annoying DX, especially with pricing module that explodes when going through pricing service.

This way, its clear that you have to keep these packages enabled from the start with a clear message.

Reply:
> Are we sure we want to force that? Couldn't the flag be used for things unrelated to the modules and that are not required?

Hmm, I actually hadn't thought of this.

My immediate thinking regarding the V2 flag was, you either opt-in for all or nothing.

So the case you are thinking of is if we are building a feature for V2, that does not require modules to be loaded right? If that's the case, wouldn't that feature be part of Medusa V1?

Reply:
I think it depends on the feature. let's go like this and see in time